versioned_docs/version-4.6/foundations/harper-architecture.md

```diff
@@ -0,0 +1,93 @@
+---
+title: Harper Architecture
+---
+
+# Harper Architecture
+
+Before diving deep into APIs and configuration, it helps to understand the big picture of how Harper works.  
+Harper uses a **three-layer architecture** designed for distributed, edge-first computing. Each layer builds on the next, letting you start simple and scale as your app grows.
+
+![](/img/v4.6/harper-architecture.png)
+
+At a high level:  
+- **Core services** handle data, networking, and files.  
+- **Extensions** layer in reusable features (REST, GraphQL, Next.js, etc.).  
+- **Applications** bring everything together to deliver user-facing functionality.  
+
+:::info
+💡 **Why it matters:** You focus on building your app, while Harper takes care of scaling, networking, and consistency behind the scenes.
+:::
+
+---
+
+## Core Services
+
+Harper ships with three essential services:
+
+- **Database** → Fast storage, queries, and transactions.  
+- **Networking** → REST/HTTP, WebSockets, MQTT, and cluster communication.  
+- **File system** → File operations and serving static assets.  
+
+Think of these as Harper’s foundation—every extension and app builds on them.
+
+---
+
+## Applications & Extensions
+
+Most of your work will happen here.  
+
+### Applications  
+Applications sit at the top layer. They’re where you implement user-facing features. Examples:  
+- A **Next.js app** served directly from Harper.  
+- An **Apollo GraphQL server** exposing APIs.  
+
+Applications don’t re-invent core logic—they declare the extensions they need.
+
+### Extensions  
+Extensions are Harper’s plug-in modules. They add reusable features to applications and can depend on each other.  
+
+| **Component**      | **Type**             | **What it adds**               | **Built on**             |
+|--------------------|----------------------|--------------------------------|--------------------------|
+| `graphqlSchema`    | Built-in Extension   | Define schemas + tables        | database                 |
+| `jsResource`       | Built-in Extension   | Custom JS resources            | database                 |
+| `rest`             | Built-in Extension   | Auto-generate REST endpoints   | networking               |
+| `@harperdb/nextjs` | Custom Extension     | Run Next.js apps               | networking, file-system  |
+| `@harperdb/apollo` | Custom Extension     | Apollo GraphQL APIs            | graphqlSchema, networking |
+
```

Comment:
This makes it seem like these are all that are available. Lets include a note that this is just a sample (relative to the arch diagram I'm noticing now) and maybe even links to all the built-ins and custom stuff.